errors/list.go

```diff
@@ -122,18 +122,28 @@ func (l *List) Append(errs ...error) {
 
 		switch e := err.(type) {
 		case hcl.Diagnostics:
-			for _, diag := range e {
-				l.errs = append(l.errs, E(diag))
+			if e != nil {
```

Comment:
now that `err` is an interface, we need to explicitly check if the underlying value is nil.

Reply:
Instead of checking if the err is nil on all cases wouldn;t it be easier to just filter out all nil errors from errs and then iterate the result ?

Reply:
Ah OK I missed the casting and was thinking about just the errs slice, sorry for the noise